### PR TITLE
[XM] Add disable switch to partial static registrar in mmp

### PR DESCRIPTION
**Pull request by chamons**

### CODE REVIEW

File: tools/mmp/driver.cs [hunk 1]
```diff
@@ -358,6 +367,7 @@ static void Main2 (string [] args)
 				{ "xamarin-framework-directory=", "The framework directory", v => { xm_framework_dir = v; }, true },
 				{ "xamarin-full-framework", "Used with --target-framework=4.5 to select XM 4.5 Target Framework", v => { IsUnifiedFullXamMacFramework = true; } },
 				{ "xamarin-system-framework", "Used with --target-framework=4.5 to select XM 4.5 Target Framework", v => { IsUnifiedFullSystemFramework = true; } },
+				{ "disable-partial-static-registrar", "Disable use of partial static registrar.", v => disable_partial_static_registrar = true },
```
Comment (rolfbjarne) [inline tools/mmp/driver.cs]: Maybe you can use the existing --registrar flag instead: if --registrar:dynamic, then don't use the partial static registrar.

i.e. only use the partial static registrar if no --registrar flag is specified.

Comment (rolfbjarne) [inline tools/mmp/driver.cs]: In that case it might make sense to make --registrar:partial work too.

Comment (chamons) [inline tools/mmp/driver.cs]: We could @rolfbjarne but my concern is that we currently mutate that value in:

```
static void Pack (IList<string> unprocessed)
{
              ... SNIP ...
			if (registrar == RegistrarMode.Default)
				registrar = App.EnableDebug ? RegistrarMode.Dynamic : RegistrarMode.Static;
			if (is_extension)
				registrar = RegistrarMode.Static;
```

And I didn't want to either stop doing that (and find every place that depended on it for now).

Comment (chamons) [inline tools/mmp/driver.cs]: The amount of mutable state in driver.cs is :(

Comment (rolfbjarne) [inline tools/mmp/driver.cs]: If you add a RegistrarMode.PartialStatic value then you could mutate to what we actually do if RegistrarMode.Default is set.

Comment (chamons) [inline tools/mmp/driver.cs]: I like that idea. Let me rework it.

**Comment by chamons** [comment]
Reworked and happier with this than my original commit.

**Comment by chamons** [comment]
Failure is a timeout in:

watchOS (BuildFailure) Show details

And appears unrelated.

**Comment by rolfbjarne** [comment]
:+1:

**Pull request merged**